staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset_test.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queueset
+
+import (
+	"context"
+	"math"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"k8s.io/apiserver/pkg/util/flowcontrol/counter"
+	fq "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing"
+	test "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing"
+	"k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing/clock"
+)
+
+type uniformScenario []uniformClient
+
+type uniformClient struct {
+	hash          uint64
+	nThreads      int
+	nCalls        int
+	execDuration  time.Duration
+	thinkDuration time.Duration
+}
+
+// exerciseQueueSetUniformScenario.  Simple logic, only works if each
+// client's offered load is at least as large as its fair share of
+// capacity.
```

Comment:
Can you expand this comment? I don't understand how this function performs a test.

Reply:
> // exerciseQueueSetUniformScenario runs a scenario based on the given set of uniform clients.
> // Each uniform client specifies a number of threads, each of which alternates between thinking
> // and making a synchronous request through the QueueSet.
> // This function measures how much concurrency each client got, on average, over
> // the intial `totalDuration` and tests to see whether they all got about the same amount.
> // Each client needs to be demanding enough to use this amount, otherwise the fair result
> // is not equal amounts and the simple test in this function would not accurately test fairness.
> // `expectPass` indicates whether the QueueSet is expected to be fair.
> // `expectedAllRequests` indicates whether all requests are expected to get dispatched.

Reply:
Added the suggested comment